feat(git): support showing diff of range of commits

AckslD opened the pull request:
If there is a range of commits, eg `00cba85..116a568` then the diff no longer only shows the diff of the first commit but of the whole range. Currently this shows the diff for each commit and not the total diff from the first to the last but that can be configured by the user with a different command.

@akinsho, I think you initially implemented this so maybe you'd be interested and I'm happy to hear any thoughts.

akinsho commented:
@AckslD can you confirm that you've tested this out locally and it works in different scenarios .e.g single and multiple commits?

AckslD commented:
@akinsho yes I have

akinsho commented:
Cool, thanks for clarifying. I've merged a few PRs in the last few weeks that have had a lot of follow-up bugs and I won't have time to test this very thoroughly, but the change looks good to me

AckslD commented:
Thanks for checking! Is there someone else who could review further if you're busy?

akinsho commented:
Think Wil is quite busy, not sure I might have time in the following days. Will try to get through your open PRs

akinsho commented:
I guess this can be quite easily tested by just reverted a plugin back a few commits and seeing that the diff shows multiple commits

akinsho commented on lua/packer.lua:
```diff
@@ -34,7 +34,8 @@ local config_defaults = {
       diff = 'log --color=never --pretty=format:FMT --no-show-signature HEAD@{1}...HEAD',
       diff_fmt = '%%h %%s (%%cr)',
-      git_diff_fmt = 'show --no-color --pretty=medium %s',
+      -- TODO maybe --no-pager should be added for all commands?
```
Re. this TODO just so it doesn't linger here forever, what would the benefit of doing this be vs the current behaviour? What advantage does it provide?

AckslD replied:
Ah right! I initially didn't get the full diff from the command and I thought it was due do this. But it was actually since only the first entry was used here:
https://github.com/wbthomason/packer.nvim/blob/master/lua/packer/display.lua#L674

Removed --no-pager again and the comment :+1:

AckslD commented:
Thanks, also no rush so take your time :)

Yeah, I wrote a small bash script which resets three of my plugins and then runs neovim and directly updates all plugins. This was very useful for quickly checking things.

akinsho commented:
@AckslD I just tested this out and got some strange results using telescope, I manually reset it back 3 commits and then hovered over the plugin name after running packer sync and the diff was making reference to 31807eef4ed574854b8a53ae40ea3292033a78ea i.e. the commit I was on before the update, rather than showing me a preview of all 3 new commits. With my cursor on each commit separately this worked as expected

AckslD commented:
@akinsho hmm, do you have any custom git settings? For me I get
```
 ✓ Updated nvim-lua/plenary.nvim: 31807ee..a3dafaa
  URL: https://github.com/nvim-lua/plenary.nvim
  Commits:
    a3dafaa feat: add svelte and astro filetype (#398) (5 days ago)
    9e6a153 docs: add instructions on how to run tests (#397) (5 days ago)
    92e6605 docs: minor changes to README.md (#399) (5 days ago)
```
after reseting three commits back and updating. Then if I press `d` on the first line I get:
````
commit a3dafaa937921a4eb2ae65820c3479ab561e9ba3
Author: Joohoon Cha <joocha0713@gmail.com>
Date:   Fri Aug 26 04:54:28 2022 -0700

    feat: add svelte and astro filetype (#398)

diff --git a/data/plenary/filetypes/builtin.lua b/data/plenary/filetypes/builtin.lua
index 57ef560..a8aaaa3 100644
--- a/data/plenary/filetypes/builtin.lua
+++ b/data/plenary/filetypes/builtin.lua
@@ -20,32 +20,34 @@ end
 return {
   extension = {
     ['_coffee'] = 'coffee',
-    ['coffee'] = 'coffee',
+    ['astro'] = 'astro',
     ['cljd'] = 'clojure',
+    ['coffee'] = 'coffee',
     ['dart'] = 'dart',
+    ['erb'] = 'eruby',
     ['ex'] = 'elixir',
     ['exs'] = 'elixir',
-    ['erb'] = 'eruby',
-    ['tres'] = 'gdresource',
-    ['tscn'] = 'gdresource',
-    ['gd'] = 'gdscript',
     ['fnl'] = 'fennel',
+    ['gd'] = 'gdscript',
     ['gql'] = 'graphql',
-    ['graphql'] = 'graphql',
     ['gradle'] = 'groovy',
+    ['graphql'] = 'graphql',
     ['hbs'] = 'handlebars',
     ['hdbs'] = 'handlebars',
     ['janet'] = 'janet',
-    ['jsx'] = 'javascriptreact',
     ['jl'] = 'julia',
+    ['jsx'] = 'javascriptreact',
     ['kt'] = 'kotlin',
     ['nix'] = 'nix',
+    ['plist'] = 'xml',
     ['purs'] = 'purescript',
-    ['rkt'] = 'racket',
     ['res'] = 'rescript',
     ['resi'] = 'rescript',
+    ['rkt'] = 'racket',
+    ['svelte'] = 'svelte',
+    ['tres'] = 'gdresource',
+    ['tscn'] = 'gdresource',
     ['tsx'] = 'typescriptreact',
-    ['plist'] = 'xml',
   },
   file_name = {
     ['cakefile'] = 'coffee',
commit 9e6a153d9a6438a8bafb63ed41c47445eecc4174
Author: Rico Sta. Cruz <rstacruz@users.noreply.github.com>
Date:   Fri Aug 26 21:54:13 2022 +1000

    docs: add instructions on how to run tests (#397)

diff --git a/TESTS_README.md b/TESTS_README.md
index e5db67a..1103ca2 100644
--- a/TESTS_README.md
+++ b/TESTS_README.md
@@ -32,6 +32,15 @@ end)
 
 The test **some test** checks that a functions output is as expected based on the input. The second test **some other test** checks that the variable **bounter** is reset for each test (as defined in the before_each block).
 
+# Running tests
+
+Run the test using `:PlenaryBustedTest <file>`. 
+
+```vimscript
+" Run the test in the current buffer
+:PlenaryBustedTest %
+```
+
 # mocking with luassert
 
 Plenary.nvim comes bundled with [luassert](https://github.com/Olivine-Labs/luassert) a library that's built to extend the built-int assertions... but it also comes with stubs, mocks and spies!
commit 92e660587d2f8f04937fb5348c1bb79cd28a2572
Author: Ahmed <48479476+XenoBino@users.noreply.github.com>
Date:   Fri Aug 26 14:53:41 2022 +0300

    docs: minor changes to README.md (#399)
    
    * doc: Added installation using packer to README.md
    
    * doc: Added neo-tree.nvim to README.md

diff --git a/README.md b/README.md
index a28dbf8..6e964b0 100644
--- a/README.md
+++ b/README.md
@@ -12,10 +12,18 @@ At the moment, it is very much in pre-alpha :smile: Expect changes to the way so
 
 ## Installation
 
+Using [plug](https://github.com/junegunn/vim-plug):
+
 ```vim
 Plug 'nvim-lua/plenary.nvim'
 ```
 
+Using [packer](https://github.com/wbthomason/packer.nvim):
+
+```
+use "nvim-lua/plenary.nvim"
+```
+
 ## Modules
 
 - `plenary.async`
@@ -92,6 +100,7 @@ end
 - [gitsigns.nvim](https://github.com/lewis6991/gitsigns.nvim)
 - [vgit.nvim](https://github.com/tanvirtin/vgit.nvim)
 - [neogit](https://github.com/TimUntersberger/neogit)
+- [neo-tree.nvim](https://github.com/nvim-neo-tree/neo-tree.nvim)
 
 ### plenary.async_lib
````

akinsho commented:
Hmm just tried again and it worked like I expect 🤔 not sure why the unusual result the first time except the difference was that the first time there were other plugin updates above and below and some issue installing a treesitter parser. I'll try and reset a few different plugins and see how it behaves

akinsho reviewed (approved):
LGTM, tested a few plugins and this seemed to work without issue on re-test

AckslD commented:
Awesome, thanks @akinsho :rocket:

ur4ltz commented:
Need add blank line before 'commit xxxxxx', this restored syntax highlight!

AckslD commented:
Hi @ur4ltz, could you clarify. Do you mean a blank line between each commit in the diff buffer?

ur4ltz commented:
Hi @AckslD! Yes!

AckslD commented:
Ah yes I see, should be fixed by #1040
